atlasdb-config/src/main/java/com/palantir/atlasdb/factory/ServiceDiscoveringAtlasSupplier.java

```diff
@@ -64,11 +64,33 @@ public ServiceDiscoveringAtlasSupplier(KeyValueServiceConfig config, Optional<Le
                         "No atlas provider for KeyValueService type " + config.type() + " could be found."
                         + " Have you annotated it with @AutoService(AtlasDbFactory.class)?"
                 ));
-        keyValueService = Suppliers.memoize(() -> atlasFactory.createRawKeyValueService(config, leaderConfig));
+        keyValueService = Suppliers.memoize(() -> pollForKeyValueService(atlasFactory));
         timestampService = () -> atlasFactory.createTimestampService(getKeyValueService());
         timestampStoreInvalidator = () -> atlasFactory.createTimestampStoreInvalidator(getKeyValueService());
     }
 
+    private KeyValueService pollForKeyValueService(AtlasDbFactory atlasFactory) {
+        long checkStartTime = System.currentTimeMillis();
+        while (true) {
+            try {
+                return atlasFactory.createRawKeyValueService(config, leaderConfig);
```

Comment:
Creating a raw C*KVS retries a bunch and has bunch of connection timeouts to wait for; it's likely this will hang for much longer than the user configured pollForDataBaseOnStartUpMillis(). There are also some edge cases where a bunch of schema migration has to happen when you start a C*KVS after some Atlas upgrades, which can take an unusually long time to startup.

I'm not sure, but it's possible the API our service manager would prefer is to have some set timeout for Atlas to wait to come up, and atlas to terminate if it goes over that limit. (e.g. if Atlas doesn't come up in 5 minutes, kill it, and give the user an alert that their service is down)
Or maybe they want the API to be the most dropwizard-style thing as possible, which is to wait forever in this loop by default, and just have failing healthchecks until we can create a KVS?